ctms/crud.py

```diff
@@ -270,23 +270,33 @@ def get_contacts_by_any_id(
 
 
 def get_all_acoustic_records_before(
-    db: Session,
-    end_time: datetime,
-    retry_limit: int = 5,
+    db: Session, end_time: datetime, retry_limit: int = 5, batch_limit=None
 ) -> List[PendingAcousticRecord]:
     """
     Get all the pending records before a given date. Allows retry limit to be provided at query time."""
-    pending_records: List[PendingAcousticRecord] = (
+    if batch_limit is None:
+        pending_records: List[PendingAcousticRecord] = (
+            db.query(PendingAcousticRecord)
+            .filter(
+                PendingAcousticRecord.update_timestamp < end_time,
+                PendingAcousticRecord.retry < retry_limit,
+            )
+            .order_by(asc(PendingAcousticRecord.update_timestamp))
```

Comment:
maybe switch this to `desc` for now

Reply:
changed this

Reply:
With these changes...we could also just give it limit 1 as envvar and spin multiple up maybe without stepping on toes too much...? 

There might be more to do to make it more parallelizable though...

Reply:
i  think limit 1  is  too few.  to actually make it parallelizable we need a lock or  partition or somesuch